### PR TITLE
Changed UoW's default TransactionScopeOption and IsolationLevel

### PR DESCRIPTION
**Pull request by cthumburu**
Changed UoW's default TransactionScopeOption to RequiresNew and IsolationLevel to ReadCommitted. Updated the unit tests.

### CODE REVIEW

File: tests/BoltOn.Tests/UoW/UnitOfWorkManagerTests.cs [hunk 1]
```diff
@@ -101,12 +101,12 @@ public void Get_WithDefaultsTwice_StartsTransactionWithDefaultsAndJoinsWithThePr
 			var result2 = sut.Get(unitOfWorkOptions);
 
 			// assert
-			var uowProviderLoggerStmt = $"About to start UoW. IsolationLevel: {IsolationLevel.Serializable} " +
+			var uowProviderLoggerStmt = $"About to start UoW. IsolationLevel: {IsolationLevel.ReadCommitted} " +
```
Comment (gokulm) [inline tests/BoltOn.Tests/UoW/UnitOfWorkManagerTests.cs]: this unit test's method name needs to be updated

File: tests/BoltOn.Tests/UoW/UnitOfWorkManagerTests.cs [hunk 2]
```diff
@@ -146,7 +146,38 @@ public void Get_WithDefaultsTwiceButSecondOneRequiresNew_StartsTransactionWithDe
...
+            var uowManagerLogger = new Mock<IBoltOnLogger<UnitOfWorkManager>>();
+            var uow = new Mock<IUnitOfWork>();
+            var uowFactory = new Mock<IUnitOfWorkFactory>();
+            var unitOfWorkOptions = new UnitOfWorkOptions();
+            uowFactory.Setup(u => u.Create(unitOfWorkOptions)).Returns(uow.Object);
+            var sut = new UnitOfWorkManager(uowManagerLogger.Object, uowFactory.Object);
+
+            // act
+            var result = sut.Get(unitOfWorkOptions);
+            unitOfWorkOptions.TransactionScopeOption = TransactionScopeOption.Required;
```
Comment (gokulm) [inline tests/BoltOn.Tests/UoW/UnitOfWorkManagerTests.cs]: 👍

File: tests/BoltOn.Tests/UoW/UnitOfWorkManagerTests.cs (hunk 1 above)
Comment (cthumburu) [reply tests/BoltOn.Tests/UoW/UnitOfWorkManagerTests.cs]: @gokulm I feel this test is redundant, and also a duplicate of next test at line #119. What do you think?

**Comment by cthumburu** [comment]
@gokulm Removed code that explicitly sets the default transactionscope to RequiresNew

File: tests/BoltOn.Tests/UoW/UnitOfWorkManagerTests.cs (hunk 1 above)
Comment (gokulm) [reply tests/BoltOn.Tests/UoW/UnitOfWorkManagerTests.cs]: @cthumburu agree. #119 test's RequiresNew can be removed though

**Comment by cthumburu** [comment]
@gokulm Addressed all the code review comments

**Comment by gokulm** [comment]
@cthumburu cool. i'm almost done with ci and cd. i'll let you know once i'm done, so that u can merge the code, and nuget pkgs will be automatically published.

**Comment by cthumburu** [comment]
@gokulm Review this PR and let me know when we can merge this PR

**Comment by cthumburu** [comment]
@gokulm Thanks for merging this.

**Comment by gokulm** [comment]
@cthumburu i merged it to test the CI and CD github actions... still there are some quirks with it, trying to fix it. once i fix it, nuget publish should be automatic.

**Pull request merged**